src/test/java/uk/gov/companieshouse/registeredemailaddressapi/unit/service/EligibilityServiceTest.java

```diff
@@ -0,0 +1,91 @@
+import uk.gov.companieshouse.registeredemailaddressapi.eligibility.impl.CompanyStatusValidation;
+import uk.gov.companieshouse.registeredemailaddressapi.eligibility.impl.CompanyTypeValidation;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.EligibilityException;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.ServiceException;
+import uk.gov.companieshouse.registeredemailaddressapi.service.CompanyProfileService;
+import uk.gov.companieshouse.registeredemailaddressapi.service.EligibilityService;
+
+@ExtendWith(MockitoExtension.class)
+class EligibilityServiceTest {
+
+
+    private List<EligibilityRule<CompanyProfileApi>> eligibilityRules;
+
+    @Mock
+    private CompanyProfileService companyProfileService;
+
+    private EligibilityService eligibilityService;
+
+    @BeforeEach
+    void init() {
+        CompanyTypeValidation companyTypeValidation = new CompanyTypeValidation(Set.of("ltd"));
+        CompanyStatusValidation companyStatusValidation = new CompanyStatusValidation(Set.of("active"));
+
+        eligibilityRules = List.of(companyTypeValidation, companyStatusValidation);
+
+        eligibilityService = new EligibilityService(eligibilityRules, companyProfileService);
+    }
+
+    @Test
+    void testWithNoErrors() throws ServiceException {
+        // GIVEN
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyStatus("active");
+        companyProfileApi.setType("ltd");
+
+        // WHEN
+        var responseBody = eligibilityService.checkCompanyEligibility(companyProfileApi);
+
+        // THEN
+        assertNotNull(responseBody);
+        assertEquals(EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE, responseBody.getEligibilityStatusCode());
+    }
+
+    @Test
+    void testInvalidCompanyStatus() throws EligibilityException, ServiceException {
+        // GIVEN
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyStatus("inactive");
+        companyProfileApi.setType("ltd");
+
+        // WHEN
+        var responseBody = eligibilityService.checkCompanyEligibility(companyProfileApi);
+
+        // THEN
+        assertNotNull(responseBody);
+        assertEquals(EligibilityStatusCode.INVALID_COMPANY_STATUS, responseBody.getEligibilityStatusCode());
+    }
+
+    @Test
+    void testInvalidCompanyType() throws EligibilityException, ServiceException {
+        // GIVEN
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyStatus("active");
+        companyProfileApi.setType("plc");
+
+        // WHEN
+        var responseBody = eligibilityService.checkCompanyEligibility(companyProfileApi);
+
+        // THEN
+        assertNotNull(responseBody);
+        assertEquals(EligibilityStatusCode.INVALID_COMPANY_TYPE, responseBody.getEligibilityStatusCode());
+    }
+
+}
```

Comment:
Is there a test for INVALID_NO_REGISTERED_EMAIL_ADDRESS_EXISTS?

Reply:
ouch, apologies for this omission, added three extra test methods